### DISCUSSION TITLE
Is there a way to exclude type-only dependencies from getting bundled together on the compiled output?

### DISCUSSION THREAD

**Discussion by IqbalLx** [nrwl/nx#17514] [General]
Hi, I am using nx to build migrasi app, you can find the repo to the project [here](https://github.com/IqbalLx/migrasi). 
The repo consists of 2 app, web app with sveltekit and node cli app. I use TRPC to connect between server endpoint to cli client, based on trpc docs they require us to import the type of trpc router from the server-side code to be used on the client-side code. Here the [link](https://trpc.io/docs/client/vanilla/setup) to the trpc docs on "how to setup their vanilla `@trpc/client`"

You can see that they mention this

> By using import type you ensure that the reference will be stripped at compile-time, meaning you don't inadvertently import server-side code into your client. For more information

if you take a look the repo, I have all trpc code in `packages/shared/trpc` folder, I also exporting the type of the router under `@migrasi/shared/trpc/routers` path, which then on the client-side code I import the type to initialize trpc client.

The problem is when I build the cli app with `nx build cli`, I notice that the server-side code also get bundled together. The functionality seems okay, the cli still make a network request to the sveltekit server endpoint and not using the server code that are bundled together. 

So my real question is, is there a way to exclude type-only dependencies from getting bundled together on the compiled output? I feel that nx somehow treat my type import as regular import and build the whole package, while I only need the type some object.

If there are some missing context please let me know. Thank you

**Comment by mfkrause** [comment]
Did you find a solution to this? I have a similar problem where a build of a Next.js application, that doesn't use TypeScript's `strict` mode is failing because it depends on the tRPC backend code (which does use strict mode) through the `import type` directive. Nx shouldn't need to depend on and check all the backend code when I'm simply importing its _types_.
